cellfinder_napari/tests/test_curation.py

```diff
@@ -81,3 +87,96 @@ def test_cell_marking(curation_widget, tmp_path):
     # Check that two .tif files are saved for both cells and non_cells
     assert len(list((tmp_path / "non_cells").glob("*.tif"))) == 2
     assert len(list((tmp_path / "cells").glob("*.tif"))) == 2
+
+
+@pytest.fixture
+def valid_curation_widget(make_napari_viewer):
+    """
+    Setup up a valid curation widget,
+    complete with training data layers and points,
+    and signal+background images.
+    """
+    viewer = make_napari_viewer()
+    image_layers = load_sample()
+    for layer in image_layers:
+        viewer.add_layer(napari.layers.Image(layer[0], **layer[1]))
+
+    curation_widget = CurationWidget(viewer)
+    viewer.window.add_dock_widget(curation_widget)
+
+    curation_widget.add_training_data()
+    # Add a points layer to select points from
+    points = Points(
+        np.array([[16, 17, 18], [13, 14, 15]]), name="selection_points"
+    )
+    # Adding the layer automatically selects it in the layer list
+    viewer.add_layer(points)
+
+    # Select the first point, and add as a cell
+    points.selected_data = [0]
+    curation_widget.mark_as_cell()
+
+    # Select the second point, and add as a non-cell
+    points.selected_data = [1]
+    curation_widget.mark_as_non_cell()
+
+    curation_widget.signal_image_choice.setCurrentText("Signal")
+    curation_widget.background_image_choice.setCurrentText("Background")
+    curation_widget.set_signal_image()
+    curation_widget.set_background_image()
```

Comment:
The pattern of setting a text value, then calling a method to set the layer seems odd to me - in other `napari` widgets the magicgui integration means that it's possible to do `widget.layer_selector.value = my_napari_layer` directly. I think this way is better (it's at least one less line of code). This comment is almost certainly beyond the scope of this PR, but thought I'd mention for info.

Reply:
Indeed, I agree. The curation widget does not use `magicgui` at all at the moment (apart from a progressbar). I've opened #129 so this is on the list.